lib/Core/TerriaError.ts

```diff
@@ -0,0 +1,32 @@
+"use strict";
+
+import i18next from "i18next";
+
+export interface TerriaErrorOptions {
+  /**  A detailed message describing the error.  This message may be HTML and it should be sanitized before display to the user. */
+  message: string;
+  /** A short title describing the error. */
+  title?: string;
+  /** The object that raised the error. */
+  sender?: unknown;
+  /** True if the user has seen this error; otherwise, false. */
+  raisedToUser?: boolean;
+}
+
+/**
+ * Represents an error that occurred in a TerriaJS module, especially an asynchronous one that cannot be raised
+ * by throwing an exception because no one would be able to catch it.
+ */
+export default class TerriaError implements TerriaErrorOptions {
```

Comment:
It seems weird to me that a class implements its own initialisation options. Is there anywhere that this property is used?

Reply:
Yeah I just did that to drag across JSDoc comments. I can get rid of it

Reply:
i got rid of it